### PR TITLE
Improved the "Button" UI component

### PR DESCRIPTION
**Pull request by uskayyyyy**
## Overview
Improved the `src/components/ui/Button.tsx` file to match it with the Figma design since it was nearly the default state generated by shadcn/ui.

## Changes
- Defined three types of variants (primary/error/cancel)
- Defined three sizes, adjusting to make the buttons' sizes as close as possible to the Figma design.
- Capitalized the file name to make it conventional.

## Note
- [x] #60 must be reviewed and merged to the base branch prior to this PR because this PR depends on it.

### CODE REVIEW

**Comment by kanta1207** [comment]
Great job! Button component is refined much better now!
Please let me hear your opinion on adding "icon" variant to Button component, 
and except for that, I think we can merge this change!

**Comment by uskayyyyy** [comment]
@kanta1207
Thank you for your prompt review! It would be better to create another component solely for icons.

**Review by kanta1207 — APPROVED** [review]
Sorry for the late response, didn't reallize there was conflict.
LGTM, great job!

File: src/components/base/parts/mobile/MobileHeader.tsx [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
       </div>
     </header>
   );
 };
+
+interface IButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  children: React.ReactNode;
+}
+
+const Button: React.FC<IButtonProps> = ({ children, ...props }) => {
```
[the remark this answers is not in the document]
Comment (uskayyyyy) [reply src/components/base/parts/mobile/MobileHeader.tsx]: If this is a component that will be re-used, it's a good idea. In this case, however, the `Button` component is private, which mean there is no `export` keyword on it, thus `className` is not necessary, I think. We can add it when it's needed. (YAGNI)

**Pull request merged**